File: 测试输出/Classes/普通对象/Person.m
//
//  Person.m
//  测试输出
//
//  Created by shoule on 15/11/20.
//  Copyright © 2015年 SaiDicaprio. All rights reserved.
//

#import "Person.h"
#import "Room.h"
#import "Car.h"
#import "Dog.h"

@interface Person()
@property (nonatomic, retain) Room *room;
@end

@implementation Person

- (void)eat{
    [NSTimer scheduledTimerWithTimeInterval:1 target:self selector:@selector(fire) userInfo:nil repeats:YES];
}

- (void)fire{
    NSLog(@"吃..");
}

- (void)meditation{
    NSLog(@"meditation");
}

- (void)sleep{
    NSLog(@"sleep");
}

@end

File: Swift中的单例/Classes/Person.swift
//
//  Person.swift
//  测试输出
//
//  Created by shoule on 16/4/29.
//  Copyright © 2016年 SaiDicaprio. All rights reserved.
//

import Foundation

class Person: NSObject {
    var name:String = ""
    
    class func sharedPerson()-> Person{
        struct PersonStruct{
            static var predicate:dispatch_once_t = 0
            static var instance:Person? = nil
        }
        
        dispatch_once(&PersonStruct.predicate, {
            PersonStruct.instance = Person()
        })
        
        return PersonStruct.instance!
    }

}
